server/api/src/rsDataObjOpen.cpp

```diff
@@ -763,10 +763,15 @@ namespace
 
         if (hierarchy_for_open.empty()) {
             try {
+                // NOLINTNEXTLINE(hicpp-signed-bitwise)
+                const auto open_for_create = 0 != (_inp.openFlags & O_CREAT);
+                const auto open_for_write = 0 != getWriteFlag(_inp.openFlags);
+                const auto operation = open_for_create  ? irods::CREATE_OPERATION
+                                       : open_for_write ? irods::WRITE_OPERATION
+                                                        : irods::OPEN_OPERATION;
```

Comment:
Consider reformatting or refactoring to use if-statements.

Chained ternary operators can hurt readability.

Reply:
Done